Feat/cookbook startupsim (#257)
rungalileo/docs-official: cookbooks/use-cases/custom-metric-startup-sim/custom-metric-startup-sim.mdx

GALILEO_PROJECT=your project name here
GALILEO_LOG_STREAM=my_log_stream

# Optional LLM configuration
LLM_MODEL=gpt-4
LLM_TEMPERATURE=0.7

# Optional agent configuration
VERBOSITY=low  # Options: none, low, high
ENVIRONMENT=development
ENABLE_LOGGING=true
ENABLE_TOOL_SELECTION=true
```
</Step>

<Step title="Start the Flask app and test out the application">
After your environment variables are set, you are all set to run the application.  The application is designed as a Flask Application, with a JavaScript frontend, and Python backend.

Run the application locally by running the following in the terminal. 

```
python web_server.py
```

Your application will be running at http://localhost:2021 — open that within your browser and start exploring. 

Try generating both "Silly" and "Serious" mode pitches.

The standard flow of the application is as follows: 
User Input → Flask Route → Agent Coordinator → Tool Chain → AI Model → Response
</Step>
</Steps>

## Create a new Galileo project

In order to set up custom metrics, we'll need a Galileo project to log evaluations to first. 

<Steps>
<Step title="Create a new project from the Galileo Console using the `New Project button`">

> If you haven't already, create a free Galileo account on app.galileo.ai. When prompted, add an organization name.  To dive right into this tutorial, you can skip past the onboarding screen by clicking on the Galileo Logo in the upper left hand corner. 

Reviewer: ```suggestion
If you haven't already, create a free Galileo account on [app.galileo.ai](https://app.galileo.ai). When prompted, add an organization name.  To dive right into this tutorial, you can skip past the onboarding screen by clicking on the Galileo Logo in the upper left hand corner. 
```

![A red arrow pointing at the Galileo logo in the upper left hand corner of the screen](/cookbooks/use-cases/custom-metric-startup-sim/galileo-logo-project.webp)

**NOTE:** you will not be able to come back to this screen again, however there are helpful instructions to getting started in the Galileo Docs.

> Create a new project by clicking on the 'new project' button on the upper right hand screen.   You will be prompted to add a project name, as well as a log stream name.

Reviewer: ```suggestion
Create a new project by clicking on the **New Project** button on the upper right hand screen.   You will be prompted to add a project name, as well as a log stream name.
```

Button labels as bold, and case match the label

</Step>

> <Step title="Getting your Galileo API Keys">

Reviewer: ```suggestion
<Step title="Get your Galileo API Keys">
```

> Once that is created.  Click on the profile icon in the upper right hand side of the page, navigate on the drop-down menu to API keys.  From the API Keys screen, select 'Create New Key'.  Save the key within your environment file with the project name and log stream name you've created.

Reviewer: ```suggestion
Once that is created.  Click on the profile icon in the upper right hand side of the page, navigate on the drop-down menu to API keys.  From the API Keys screen, select **Create New Key**.  Save the key within your environment file with the project name and log stream name you've created.
```

Try generating both "Silly" and "Serious" mode pitches.

The standard flow of the application is as follows: 
User Input → Flask Route → Agent Coordinator → Tool Chain → AI Model → Response
</Step>
</Steps>

> ## Creating a custom LLM-as-a-Judge metric in Galileo

Reviewer: ```suggestion
## Create a custom LLM-as-a-Judge metric in Galileo

```

<Steps>

> <Step title="Adding metrics to your log stream in Galileo">

Reviewer: ```suggestion
<Step title="Add metrics to your log stream in Galileo">
```

Navigate to your project home inside of Galileo. 

Look for the name of your project, and open it up to the log stream you've got your traces in.

> Click on the "trace" view and see your most recent runs listed below, it should look something like below.

Reviewer: ```suggestion
Click on the **Trace** view and see your most recent runs listed below, it should look something like below.
```

![A view of log streams within the Galileo Console](/cookbooks/use-cases/custom-metric-startup-sim/galileo-logo-project.webp)

> From this view, navigate to the upper right hand side of your screen and click on the `Configure Metrics` button.  A side panel should appear with a set of different metrics from you to choose from. 

Reviewer: ```suggestion
From this view, navigate to the upper right hand side of your screen and click on the **Configure Metrics** button.  A side panel should appear with a set of different metrics from you to choose from. 
```

![A gif showing where to see all metrics within Galileo](/cookbooks/use-cases/custom-metric-startup-sim/configure-metrics.gif)
</Step>

> <Step title="Adding custom metrics">

Reviewer: ```suggestion
<Step title="Add custom metrics">
```

That's where custom metrics come in.

> Once in this panel, navigate to the "Create Metric" in the upper right hand corner of your screen, and select "LLM-as-a-Judge" Metric.

Reviewer: ```suggestion
Once in this panel, navigate to the **Create Metric** in the upper right hand corner of your screen, and select **LLM-as-a-Judge** Metric.
```

![A screenshot with a red arrow pointing to the Create Metric button in the upper right hand corner of the screen.](/cookbooks/use-cases/custom-metric-startup-sim/LLM-as-judge-Metric.webp)
</Step>

> <Step title="Creating your own LLM-as-a-Judge prompt">

Reviewer: ```suggestion
<Step title="Create your own LLM-as-a-Judge prompt">
```

3. CULTURAL RELEVANCE
   - [ ] References are current and timely
   - [ ] Captures current startup culture trends
   - [ ] Buzzwords are accurately parodied
   - [ ] Industry-specific knowledge is evident
4. NARRATIVE COHERENCE
   - [ ] Story follows internal logic
   - [ ] Pivots make sense within context
   - [ ] Character/voice remains consistent
   - [ ] Plot points connect logically
5. ORIGINALITY
   - [ ] Avoids overused startup jokes
   - [ ] Contains unique elements
   - [ ] Offers fresh perspective
   - [ ] Surprises the audience
6. TECHNICAL ACCURACY
   - [ ] Startup concepts are correctly parodied
   - [ ] Industry terminology is used appropriately
   - [ ] Business concepts are accurately mocked
   - [ ] Technical details are correctly referenced
 Answer TRUE only if ALL of the following conditions are met:
   - [ ] At least 80% of all criteria are rated TRUE
   - [ ] No critical criteria (Satire Effectiveness, Humor Consistency) are rated FALSE
   - [ ] Content would be considered funny by the target audience
   - [ ] Satire successfully achieves its intended purpose
   - [ ] Content maintains appropriate tone throughout
```

> When added, press 'save' then test your metric.  The evaluation prompt will then be generated for you to see within a preview window. 

Reviewer: ```suggestion
When added, press **Save** then test your metric.  The evaluation prompt will then be generated for you to see within a preview window. 
```

</Step>

> <Step title="Testing your metric">

Reviewer: ```suggestion
<Step title="Test your metric">
```

> From within the Custom Metric UI, select Test Metric. 

Reviewer: ```suggestion
From within the Custom Metric pane, select **Test Metric**. 
```

![A screenshot showing where to test the metric within the Custom Metric UI](/cookbooks/use-cases/custom-metric-startup-sim/test-custom-metric.webp)

> Take the output from an earlier run, and paste it in the output section of the 'test metric' page, and check the response.

Reviewer: ```suggestion
Take the output from an earlier run, and paste it in the output section of the **Test Metric** page, and check the response.
```

> ![A screenshot showing where to test the metric within the Custom Metric UI](/cookbooks/use-cases/custom-metric-startup-sim/tested-metric.webp)

Reviewer: Is this alt text correct?

Continue tweaking the prompt until you have a metric that you feel confident with — the goal isn't to have a metric that is perfect 100% of the time, but helps you determine what "good" looks like.

Have examples of what a subject matter expert would consider to be "good" and "bad" to test your metric for success.
</Step>

> <Step title="Adding your Custom Metric">

Reviewer: ```suggestion
<Step title="Add your Custom Metric">
```

> Once tested, your metric will appear in the list of available metrics.  Click on "Configure Metrics" and flip the toggle the metric you've created, on. From there, it can be used to assess future LLM outputs across runs and sessions—giving you visibility into quality over time.

Reviewer: ```suggestion
Once tested, your metric will appear in the list of available metrics.  Click on **Configure Metrics** and flip the toggle the metric you've created, on. From there, it can be used to assess future LLM outputs across runs and sessions—giving you visibility into quality over time.
```
